### PR TITLE
Lavaland dwarfs 4(?): He does it for free edition.

### PR DESCRIPTION
**Pull request by Shadowlight213**
:cl: MMMiracles,Iamgoofball,Ma44,Shadowlight213,Militaires
add: Lavaland dwarves
/:cl:

I was bored and decided to see if I can finish this.
Currently compiles, 0 hours of testing performed.

### CODE REVIEW

**Comment by MoreRobustThanYou** [comment]
Squash the commits fuckwad.

**Comment by MoreRobustThanYou** [comment]
learn how to do a ```git rebase -i ````

File: code/game/atoms.dm
Comment (ACCount12) [inline code/game/atoms.dm]: the idea of every single atom having those procs is both amusing and terrifying to me.

**Comment by fluxcapacitor1337** [comment]
NICE

**Comment by fluxcapacitor1337** [comment]
is this a playable race? i want to play as a dwarf every round

**Comment by Kawaii-Big-Boss** [comment]
Pretty sure it's like ashliggers, where you go to lavaland to find them.
You can dorf every round assuming there's a way to get to the station.

**Comment by fluxcapacitor1337** [comment]
i dont want no lavaland shenanigans i want to hang out with my fellow dwarves in bar plotting atrocities

**Comment by Jordie0608** [comment]
capture

**Comment by Shadowlight213** [comment]
to be fair, that's on the map, which isn't compiled by default

**Comment by davethwave** [comment]
It would be nice if we could merge this in my opinion. The amount of testing and change this idea has overcome is amazing.

**Comment by MoreRobustThanYou** [comment]
HMMMM DOES IT COMPILE

**Comment by Kawaii-Big-Boss** [comment]
We must merge this immediately.

**Comment by Shadowlight213** [comment]
fuck

**Comment by Shadowlight213** [comment]
:tada:

**Comment by Kawaii-Big-Boss** [comment]
You did it, Shadowlight.
You are the hero the server needed, but not the one it deserved.

**Comment by fluxcapacitor1337** [comment]
it is of utmost importance to have this as a station race similar to moths and humans and skeletons

**Comment by MMMiracles** [comment]
the wiki is still up for anyone wanting to see the actual content without skimming through 100 files

https://tgstation13.org/wiki/Lava_Dwarf

File: code/game/atoms.dm
Comment (optimumtact) [reply code/game/atoms.dm]: goof code

**Comment by Shadowlight213** [comment]
note to self. bar will create 5 sheets of metal first time, but will only add one to the stack if one is there.

**Comment by Shadowlight213** [comment]
Another note. The fort in a box things don't clear out walls when they deploy and spawn with vacuum.

**Comment by Shadowlight213** [comment]
<dwarves_are_a_mistake> Halt that dwarf pr
<dwarves_are_a_mistake> It's not good in its current state design wise
<dwarves_are_a_mistake> I made a huge mistake adding fortinaboxes
<dwarves_are_a_mistake> Don't merge it until he switches it to migrant waves again instead of shitty dorm spawners
<dwarves_are_a_mistake> Or optimally stop ignoring me about why dwarf design is bad
<dwarves_are_a_mistake> https://github.com/tgstation/tgstation/pull/27761#issuecomment-337077171
<dwarves_are_a_mistake> Alongside the issues listed there
<dwarves_are_a_mistake> Smithing needs to be made more OPP, crossbows are really buggy and bolt smithing doesn't work

posting this as a self memo since i also think fort in a box is bad design.
Also remonder to update the changelog with credit to all

**Comment by AdamElTablawy** [comment]
summary of bugs:

!!jobbanned players can play as dwarves!!

stills have no sprite

bar molds will produce a stack of 5 when first hit but if a stack of the same type is on top of them will only add 1 to the additional stack

drinking will make a dwarf permanently fat for the rest of the round (needs testing, only happened to me, didn't respawn to see if it happened again)

room-in-a-box don't actually clear out rock walls when they spawn and have odd range detection, usually have no pressure


crossbows just don't work


not a bug but QoL stuff:

each 1 individual ore must be put into the smelter, by hand, which is extremely slow, probably needs the ability to offload via dump

1 ore = 75u of the material in a bucket = incredible amounts of ore available even early on with almost no mining required, buckets EVERYWHERE-probably needs to have some sort of internal reagent storage that dwarves can retrieve from rather than covering everything in buckets

no actual source of leather-neither butchered cows or chickens will provide and lavaland animals obviously do nothing

about five minutes of mining is sufficient to power like fifteen dwarves for the rest of the round

adamantine incredibly easy to come by, seems to be more frequent than diamond, which is odd since adamantine is far more effective than diamond is

when you put a mold on the anvil you can only fill it with 25u and once you use it for something it automatically comes off the anvil and you have to juggle with your fucking bucket and hammer again to repeat the process which ultimately slows everything down- solution is to just let the fucking mold stay on the anvil when empty so you can refill

**Comment by fluxcapacitor1337** [comment]
it is of utmost importance to have this as a station race similar to moths and humans and skeletons

**Comment by AdamElTablawy** [comment]
it is of the utmost importance that you fuck riiiiight off

**Comment by Shadowlight213** [comment]
@AdamElTablawy turns out cats are a source of leather. I just deleted them that round not knowing that.

Iamgoofball - Today at 7:44 PM
@Shadowlight213 there's a reliable source of leather
1. make a breeding pen for cats
2. harvest and butcher cats, use knife on hide
3. wet hides at the puddle in the grow room
4. tan hides at the tanner you made from wood

Still sprite and jobban is fixed.
Fort in a box is planned to be scrapped.
crossbows probably disabled too until fixed.
Fat thing happened to me too.
Goof says no to the pooled resources thing. Said make a storeroom. Might map in a tiny example storeroom.

**Comment by AdamElTablawy** [comment]
goof is aware that there is going to be like, 15 million fucking buckets scattered everywhere because mining even one goddamn ore tile yields like 5 ore and that leads to 5 buckets, right?

i understand the idea of 'make a storeroom' but there's 'storeroom' and there's 'rightclick pile of death three minutes into the round'

**Comment by calzilla1** [comment]
I can't wait for the syndicate and dorfs to team up to slay the ligger menace

**Comment by Shadowlight213** [comment]
Iamgoofball - Today at 8:36 PM
@Shadowlight213 re, too much trash, lavaland has lava, use it to destroy trash

**Comment by AdamElTablawy** [comment]
less a matter of too much trash and more that even a mildly productive group of dwarves will have about sixty buckets of material they never actually use because there's so much to fucking smelt

need i remind you the ore bag fills up very quickly and caps out at 50

and that means 50 fucking buckets alone from one trip

**Comment by tortellinitony** [comment]
Completely ignored my brave attempt at dwarves. Frankly, I'm quite offended.

**Comment by optimumtact** [comment]
any lavaland ruin that requires at least 7 or more ghosts from roundstart cannot in my opinion function at our current player levels without stealing the entirety of the crew away

**Comment by AdamElTablawy** [comment]
it works fine with just two people honestly, it's just the more the merrier

**Comment by praisenarsie** [comment]
@fluxcapacitor1337 go back to hippie, you can ask them to make it a playable race once the pr is merged here and mirrored

**Comment by ZeWaka** [comment]
I'm very excited! These seem like such a great and fun feature.

**Comment by megatiger78** [comment]
There's no reason to have this as a playable race when it could just be merged into the dwarfism mutation

**Comment by FantasticFwoosh** [comment]
Now for your scheduled dead player feedback, who hasn't played dwarves since it was put on hiatus awaiting this PR.

- Fort in a box is a good design to fill out rooms quickly, alternatively deployable machines & 'workshops' would be a better alternative if you don't like it because building & refining metal sheet by sheet & assigning rooms to be ash safe is tedious.

- You can make leather from wet hides over grilles, (bonfire + rods + probably a torch which is pre-supplied with the fortress). no idea what you're yapping about a 'tanner' for like its going to replace anything.

- I agree with the sentiment over ore bucket spam & alcohol drink spam, Goof discussed having them auto dissapear like slime cores when empty but nothing came of it, big ore molten smelter (extra points for being built over lava?) would be welcome.

- [Oh boy alcohol is a whole other kettle of fish, goof defends it to the ends of the earth, i've already explained it, just remember this is the stubborn guy you're devving it with, goof is too lazy to even code drink recipes and insists on ale when regular alcohol seemingly works.](https://tgstation13.org/phpBB/viewtopic.php?f=10&t=12493&p=326454&hilit=dwarf#p326454)

- Miasma control is still abominable, a dedicated cat pen (corral ranch?) would help as dwarves tread bloody footprints wading through guts and most players don't pay attention then completely hinder the rest of the fortress.

- I have a bunch of sprites & shit for stuff like pig-tail plants (used for fibre & booze) just hit me up if you need anything

~Fin~

**Comment by Buggy123** [comment]
What, no FUCK tag? With 100+ commits on a former Goofball PR?

**Comment by Shadowlight213** [comment]
@tortellinitony gihub says you deleted your fork or something, so i wasnt able to pull from your pr.

**Comment by ma44** [comment]
Adamantine does have the same spawn rate as diamonds, so it's probably just byond rng. 

Crossbows are apparently /obj/item/gun/ballistic/automatic/speargun but with a crossbow mag type, so the item should work.

**Comment by BeeSting12** [comment]
Stills have no sprite.

**Comment by Militaires** [comment]
i posted the still sprites like 20 times in discord for goof and 20 times elsewhere 
still
also thanks alot for crediting me for the dwarf, smelter, anvil, and mold sprites

**Comment by Militaires** [comment]
GOD DAMMIT

**Comment by GunHog** [comment]
Emperor bless you for bringing this back!

**Comment by Shadowlight213** [comment]
Stupid tgs3 ui. I thought i had updated the testmerge with the stills fix yesterday.

**Comment by Buggy123** [comment]
Adamantite is eaten by the ORM but is not available in the menu, seems like it simply vanishes.

**Comment by Bawhoppen** [comment]
Does this version remove the dumb stuff that goof refused to remove? (such as infinite dwarf spawning, or being able to make tools out of literally ANY reagent)

**Comment by GunHog** [comment]
B-but...milk swords....:(

**Comment by Armhulen** [comment]
@FantasticFwoosh didn't read lol

File: code/datums/ruins/lavaland.dm
Comment (KorPhaeron) [inline code/datums/ruins/lavaland.dm]: This needs to not have a cost of 0. There is no reason for this to spawn every single round, even while testmerged. It will just run it into the ground

**Review by imtakingabreakdontatme — CHANGES_REQUESTED** [review]

File: code/game/objects/items.dm
Comment (ACCount12) [inline code/game/objects/items.dm]: well, this is better than every single atom having this proc, but still

File: code/datums/ruins/lavaland.dm
Comment (FantasticFwoosh) [reply code/datums/ruins/lavaland.dm]: This is goof's intention, [he basically told changelingrain otherwise](https://github.com/tgstation/tgstation/pull/29833#issuecomment-322025359) last time he was pulled up on it without compromising on ideas to make the changes to LL (adamantine spawning for instance) less dwarf centric on attitudes of not giving station mining or pre-techwebs RND any minerals or use from it either.

**Comment by FantasticFwoosh** [comment]
@Bawhoppen i agree, else this is going to get put back in the attic with the rest of goof's repressed pr's for usually the same reason.

@Buggy123 the ORM has special lists, you'll need to check whether its compatible with on @Shadowlight213 's side. ORM disks could be used to update it like how you can upload the ORM to print abductor sheets but that's mostly for alloys.

Anyway i think goof left it out because he made it have no intentional use on the station "**_Muh dwarf resources_**"

File: code/datums/ruins/lavaland.dm
Comment (Buggy123) [reply code/datums/ruins/lavaland.dm]: Okay, fair enough. But that's stupid, so let's ignore him and change it anyway.

**Comment by Shadowlight213** [comment]
I have school stuff atm, and goof said he was gonna pr something to my branch. Ill work on it more this weekend

**Comment by Buggy123** [comment]
As per the bucket spam issue, would it be possible to merge reagent containers on the same tile, a-la sheet stacks? So instead of buckets, you have a... !!Magic Bucket!!, which can hold say, 10 ore worth of liquid, and if two magic buckets with the same reagents are on the same tile, they merge unless one of the buckets is full/would be full.

Another problem - is it intentional that ore-to-reagent ratios are off? It's 1 ore to 75 reagents, while the normal ratio is 1-20.

**Comment by Profakos** [comment]
@FantasticFwoosh  Adamantine vanishes from the ORM, because ORM moves the materials from ores into itself, and adamantine's material is actually Titanium.

**Comment by FantasticFwoosh** [comment]
@Profakos it was a while ago but i was [already aware of this](https://github.com/tgstation/tgstation/pull/27761#issuecomment-305897940), i had just forgotten it was a issue and i also posted a resolution.

`materials = list(MAT_TITANIUM=MINERAL_MATERIAL_AMOUNT)`

Just create a new adamantine entry, done but im pretty sure Goof did this on purpose to block mining from gaining mineral points & it being processed by RnD since they were not bothered to fix it after it'd been pointed out to be a "Dwarf Only Feature". The Goofspiracy to hijack lavaland coding development by inundating it with dwarves continues.  △ 👁

**Comment by Profakos** [comment]
Ah, sorry, I forgot about the old discussion.

**Comment by Buggy123** [comment]
Oh, so THAT'S why nothing happened when i spilled a bucket of plasma.

**Comment by FantasticFwoosh** [comment]
Yeah @Buggy123 can a maintainer put that up to review so we can whip goof's ass on creepcoding please.

**Comment by Buggy123** [comment]
Hey @Iamgoofball while you're at it don't forget to remove secborgs and nerf science as well. Creepcoding gonna creep.

**Comment by FantasticFwoosh** [comment]
Nearly a week later with still required changes, it being paid for would have been neater, cleaner & faster than doing it for free.

**Comment by Bawhoppen** [comment]
I cannot fathom that Fwoosh is a real person

**Comment by FantasticFwoosh** [comment]
Im a real human bean

Face it you walked right into that one.

File: _maps/RandomRuins/LavaRuins/lavaland_surface_dorffort.dmm
Comment (ShizCalev) [inline _maps/RandomRuins/LavaRuins/lavaland_surface_dorffort.dmm]: TGM

**Review by ShizCalev — CHANGES_REQUESTED** [review]

File: _maps/RandomRuins/LavaRuins/lavaland_surface_dorfwagon.dmm
Comment (ShizCalev) [inline _maps/RandomRuins/LavaRuins/lavaland_surface_dorfwagon.dmm]: TGM

File: _maps/templates/dwarf/brewery.dmm
Comment (ShizCalev) [inline _maps/templates/dwarf/brewery.dmm]: TGM

File: _maps/templates/dwarf/dining_hall.dmm
Comment (ShizCalev) [inline _maps/templates/dwarf/dining_hall.dmm]: TGM

File: _maps/templates/dwarf/dorm.dmm
Comment (ShizCalev) [inline _maps/templates/dwarf/dorm.dmm]: TGM

File: _maps/templates/dwarf/farm.dmm
Comment (ShizCalev) [inline _maps/templates/dwarf/farm.dmm]: TGM

File: _maps/templates/dwarf/smeltery.dmm
Comment (ShizCalev) [inline _maps/templates/dwarf/smeltery.dmm]: TGM

File: _maps/templates/dwarf/throne_room.dmm
Comment (ShizCalev) [inline _maps/templates/dwarf/throne_room.dmm]: TGM

**Review by ShizCalev — CHANGES_REQUESTED** [review]

**Comment by Shadowlight213** [comment]
still waiting for that pr to the branch goof promisedwill work on this this weekend

**Comment by GunHog** [comment]
> still waiting for that pr to the branch goof promised
will work on this this weekend

I have not seen Goof around Github in a while. I think it is best that you clean up the PR code yourself

**Comment by optimumtact** [comment]
@tgstation/commit-access unless one of you want to pick this up and review with shadowlight I'm closing this in 48 hours

**Comment by FantasticFwoosh** [comment]
@optimumtact you should honestly close the next dwarf PR if its not made out of fresh commits with the outstanding required changes here and legacy of bad commit titles & problems, this PR was just doled out to @Shadowlight213 just to get Goof some validity it seems since they can't actually work on any breakthrough developments themselves about it.

Change to policy that coders should not give PR publishing responsibility to those who can't actually work on themselves please, @Iamgoofball seems to be abusively using @Shadowlight213 to bypass the new commit/fix tracker which would lead to his PR's being immediately closed.

**Comment by optimumtact** [comment]
oh do shutup fwoosh

**Comment by DoctoreMad** [comment]
He's right though

**Comment by Shadowlight213** [comment]
Will work on this later.
Also, no one told me to make this pr, like you seem to think.
I find dorfs a fun idea, and I think it can work.

**Comment by GunHog** [comment]
Please do not give up on Dwarves! Lots of players are rooting for you!

**Comment by FantasticFwoosh** [comment]
Nothing personal @Shadowlight213 but @Iamgoofball (a grown man even though he may not act like it) giving attempting to pass off their own responsibility to fix their broken shit to a kid still in school doesn't seem like a very mature or appropriate thing to do.

**Comment by duncathan** [comment]
what the fuck is fwoosh talking about lmaooo
goofball gave up on dwarves because he knows exactly what's gonna happen to them and it's not gonna matter who PRs them
he did not tell anyone to reopen dwarves

**Comment by Bawhoppen** [comment]
Fwoosh is literally a master at baiting people

If they weren't doing it intentionally, they'd definitely have replied to someone who called them out for bad posts atleast once

**Comment by FantasticFwoosh** [comment]
If you say i am a furious master-baiter @Bawhoppen then i suppose in your opinion, i am.

I am glad this PR was not re-attempted again between the time checking now and my last post here.

**Comment by ishitbyabullet** [comment]
when can we get dorfs back, fams

**Comment by epochayur** [comment]
\>old PR gets updated
\>get excited
\>it's just the branch being deleted

**Pull request closed without merging**
